Integrate DHCP/DNS into Smart Proxies (#1440)
theforeman/foreman-documentation: guides/common/modules/con_managing-dhcp-using-smart-proxies.adoc

[id="Managing_DHCP_Using_Smart_Proxies_{context}"]
= Managing DHCP Using {SmartProxies}

{SmartProxies} can manage DHCP within their network.
You can use the DHCP modules for {SmartProxy} to query for available IP addresses, add new, and delete existing reservations.
Note that your {SmartProxy} cannot manage subnet declarations.
ifdef::foreman-el,foreman-deb,katello[]
For more information on managing subnet declarations, see https://github.com/theforeman/puppet-dhcp[puppet-dhcp].

> endif::[]

Reviewer: This ifdef block really belongs in a part of the manual, if any at all. Mostly because we don't have instructions about how to map the DHCP options in that module to what you can use in the installer. Right now I'm leaning to dropping it

> endif::[]

Reviewer: This point is still open.

> endif::[]

Reviewer: I think so yes.

> endif::[]

Author: Removed.

[id="Managing_DHCP_Using_Smart_Proxies_{context}"]
= Managing DHCP Using {SmartProxies}

{SmartProxies} can manage DHCP within their network.
You can use the DHCP module of {SmartProxy} to query for available IP addresses, add new, and delete existing reservations.
Note that your {SmartProxy} cannot manage subnet declarations.
ifdef::foreman-el,foreman-deb,katello[]
For more information on managing subnet declarations, see https://github.com/theforeman/puppet-dhcp[puppet-dhcp].
endif::[]

The DHCP module can manage various available DHCP providers:

* `dhcp_infoblox` - Infoblox DHCP server
* `dhcp_isc` - ISC DHCP server over OMAPI
* `dhcp_remote_isc` - ISC DHCP server over OMAPI with leases mounted through networking

> * `dhcp_libvirt` - dnsmasq DHCP via libvirt API

Reviewer: ```suggestion
ifndef::satellite[]
* `dhcp_libvirt` - dnsmasq DHCP via libvirt API
endif::[]
```

Author: Applied for libirt.

dhcp_native_ms is orcharhino only:
```
git grep -i -n -C 1 assembly_configuring-smart-proxy-on-microsoft-windows.adoc
guides/doc-Installing_Proxy/master.adoc-28-ifdef::orcharhino[]
guides/doc-Installing_Proxy/master.adoc:29:include::common/assembly_configuring-smart-proxy-on-microsoft-windows.adoc[leveloffset=+1]
guides/doc-Installing_Proxy/master.adoc-30-endif::[]
```

[id="Managing_DHCP_Using_Smart_Proxies_{context}"]
= Managing DHCP Using {SmartProxies}

> {Project} can integrate with a DHCP service using {SmartProxies}.

Reviewer 2: Sorry to say this, but I'd prefer it the other way around - the singular incl. in the title, ID and filename.

Reviewer: It was the inconsistency that bothered me. Singular is fine too.

Author: Changed title, anchor, and file name to singular for both dhcp and dns.

> {Project} can integrate with a DHCP service using your {SmartProxy}.

Reviewer: Perhaps a writer could review this, but the title is plural (SmartProxies) and this sentence is singular. Should it be consistent? `your {singular}` implies there is one while you can use any number of them.

Author: Changed to "SmartProxies" attribute.

Author: Please close this thread if you agree @Lennonka

> The {SmartProxy} has multiple providers, allowing you to integrate with your existing DHCP infrastructure or deploy a new one.

Reviewer: Should be generic?
```suggestion
A {SmartProxy} has multiple providers, allowing you to integrate with your existing DHCP infrastructure or deploy a new one.
```

Also, would it make sense to mention that one `{SmartProxy}` can only use one provider active at a time?

Author: Applied. IMHO only one service is a given due to line 4.

Note that your {SmartProxy} cannot manage subnet declarations.
ifdef::foreman-el,foreman-deb,katello[]
For more information on managing subnet declarations, see https://github.com/theforeman/puppet-dhcp[puppet-dhcp].
endif::[]

The DHCP module can manage various available DHCP providers:

> * `dhcp_infoblox` - Infoblox DHCP server

Reviewer: I like what you did with with DNS and link to the "using ..." part. Could you do that here too?

Author: Applied!

> * `dhcp_isc` - ISC DHCP server over OMAPI

Reviewer: This should link to "Configuring DNS, DHCP, and TFTP on ..."

Author: Applied.

> * `dhcp_remote_isc` - ISC DHCP server over OMAPI with leases mounted through networking

Reviewer: This should link to the "Configuring ... with external DHCP" section since that documents it now.

Author: Applied.

[id="Managing_DHCP_Using_Smart_Proxies_{context}"]
= Managing DHCP Using {SmartProxies}

> {Project} can manage DHCP service using your {SmartProxy}.

Reviewer: I'd say `integrate with` is better. It can manage ISC DHCP but all others only talk using APIs.
```suggestion
{Project} can integrate with a DHCP service using your {SmartProxy}.
```

Author: Applied.

The {SmartProxy} has multiple providers, allowing you to integrate with your existing DHCP infrastructure or deploy a new one.
You can use the DHCP module of {SmartProxy} to query for available IP addresses, add new, and delete existing reservations.
Note that your {SmartProxy} cannot manage subnet declarations.
ifdef::foreman-el,foreman-deb,katello[]
For more information on managing subnet declarations, see https://github.com/theforeman/puppet-dhcp[puppet-dhcp].
endif::[]

The DHCP module can manage various available DHCP providers:

> * `dhcp_infoblox` - For more information, see {ProvisioningDocURL}Using_Infoblox_as_DHCP_and_DNS_Providers_provisioning[Using Infoblox as DHCP and DNS Providers] in _{ProvisioningDocTitle}_.

Author: Reading "long term", I'd say that this could be done in a follow-up PR.

> * `dhcp_infoblox` - For more information, see {ProvisioningDocURL}Using_Infoblox_as_DHCP_and_DNS_Providers_provisioning[Using Infoblox as DHCP and DNS Providers] in _{ProvisioningDocTitle}_.

Reviewer: Exactly.

* `dhcp_isc` - ISC DHCP server over OMAPI.
For more information, see {InstallingSmartProxyDocURL}configuring-dns-dhcp-and-tftp_smart-proxy[Configuring DNS, DHCP, and TFTP on {SmartProxyServer}] in _{InstallingSmartProxyDocTitle}_.

> * `dhcp_remote_isc` - ISC DHCP server over OMAPI with leases mounted through networking

Reviewer: ```suggestion
* `dhcp_remote_isc` - ISC DHCP server over OMAPI with leases mounted through networking.
```

Author: Applied.
